### PR TITLE
[HDF5] Update to 1.14.0

### PR DESCRIPTION
**Pull request by mkitti**
I doubt this is going to work.

1. <strike>Do we know how to decompress .conda files?</strike> We do know how: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/297
2. msys2 and conda-forge are using openssl v3 (I got conda-forge to backport for us)
3. msys2 picked up a curl and openssl dependency

### CODE REVIEW

**Comment by mkitti** [comment]
Looks like libsz.so.2 is missing for x86-64-linux-gnu.

conda-forge added a libaec dependency which provides libsz.so.2
https://github.com/conda-forge/hdf5-feedstock/pull/179

Maybe we should try to build libaec here?
https://gitlab.dkrz.de/k202009/libaec

**Comment by mkitti** [comment]
wat? That actually kind of worked...

**Comment by mkitti** [comment]
Hmm... I'm very suspicious about this passing now.

I think we should consider moving to our own libaec though:
https://github.com/JuliaPackaging/Yggdrasil/pull/6288

**Comment by giordano** [comment]
Is this good to go now?

File: H/HDF5/build_tarballs.jl [hunk 1]
```diff
@@ -84,6 +101,7 @@ dependencies = [
     Dependency("Zlib_jll"),
     Dependency("OpenSSL_jll"; compat="1.1.10"),
     Dependency("LibCURL_jll"),
+    Dependency("libaec_jll"; compat="1.0.6"),
```
[the remark this answers is not in the document]
Comment (mkitti) [reply H/HDF5/build_tarballs.jl:104]: No, not just Linux. In https://github.com/JuliaPackaging/Yggdrasil/pull/6274/commits/3dbe2213263b9a544e39b4c8ee654774351b747e I had to remove libaec from msys2. The msys2 HDF5 package also requires libaec:

https://packages.msys2.org/package/mingw-w64-x86_64-hdf5

[the remark this answers is not in the document]
Comment (mkitti) [reply H/HDF5/build_tarballs.jl:104]: libaec is needed for SZIP support and SZIP is enabled in their builds.

https://github.com/msys2/MINGW-packages/blob/ce86a20a72de5d3b64c58c953ef90c95f68b99d5/mingw-w64-hdf5/PKGBUILD#L90

**Comment by mkitti** [comment]
> Is this good to go now?

I think so, but I would like to test it. Is there an easy way to test it other than building it locally?

**Comment by giordano** [comment]
Publish to your own repo?

**Comment by mkitti** [comment]
I'm seeing some notes such as this:

```
┌ Warning: Dependency libaec_jll does not have a mapping for artifact libaec for platform i686-w64-mingw32-libgfortran3-cxx03
└ @ BinaryBuilderBase ~/.julia/packages/BinaryBuilderBase/g3DQ3/src/Prefix.jl:752
┌ Warning: Linked library libcrypto-3.dll could not be resolved and could not be auto-mapped
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:383
┌ Warning: Linked library libsz.dll could not be resolved and could not be auto-mapped
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:383
┌ Warning: Dependency libaec_jll does not have a mapping for artifact libaec for platform x86_64-w64-mingw32-libgfortran3-cxx03
└ @ BinaryBuilderBase ~/.julia/packages/BinaryBuilderBase/g3DQ3/src/Prefix.jl:752
┌ Warning: Linked library libcrypto-3-x64.dll could not be resolved and could not be auto-mapped
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:383
┌ Warning: Linked library libsz.dll could not be resolved and could not be auto-mapped
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:383
```

File: H/HDF5/build_tarballs.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (giordano) [reply H/HDF5/build_tarballs.jl:104]: I mean, you literally built the package only for Linux: https://github.com/JuliaPackaging/Yggdrasil/blob/010f6f2a395ae57e0606699d71db3b5fc9414138/L/libaec/build_tarballs.jl#L30

[the remark this answers is not in the document]
Comment (mkitti) [reply H/HDF5/build_tarballs.jl:104]: Oh... we should fix that.

[the remark this answers is not in the document]
Comment (mkitti) [reply H/HDF5/build_tarballs.jl:104]: See https://github.com/JuliaPackaging/Yggdrasil/pull/6299

**Comment by mkitti** [comment]
The libcrypto-3-x64.dll is a real problem, I think. We may also need to still pull zlib in since `dlopen` is looking for `zlib1`.

**Comment by giordano** [comment]
> The libcrypto-3-x64.dll is a real problem, I think.

Where's that supposed to be?

> We may also need to still pull zlib in since dlopen is looking for zlib1.

Hasn't it always been there?  If it's in the same directory it should be found automatically.

**Comment by mkitti** [comment]
I removed zlib from msys2 thinking our Zlib_jll would take care of it.

**Comment by mkitti** [comment]
libcrypto-3 comes from OpenSSL 3

**Comment by mkitti** [comment]
It's not pretty, but I think this works now. Note that these seem to require avx.

```
┌ Warning: Minimum instruction set detected for lib/libhdf5.so.310.0.0 is avx, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
┌ Warning: Minimum instruction set detected for bin/libhdf5-310.dll is avx2, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
┌ Warning: Minimum instruction set detected for bin/libhdf5_hl-310.dll is avx2, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
┌ Warning: Minimum instruction set detected for bin/libhdf5_tools-310.dll is avx2, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
┌ Warning: Minimum instruction set detected for bin/libssl-3-x64.dll is avx2, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
┌ Warning: Minimum instruction set detected for bin/zlib1.dll is avx2, not x86_64 as desired.
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/0CUml/src/Auditor.jl:311
```

To get this to work, I pulled zlib and openssl v3 from msys2.

**Comment by mkitti** [comment]
This post was superseded by cross-compiled source builds  in #6551

**Pull request closed without merging**